cvat/apps/engine/media_extractors.py

```diff
@@ -905,31 +920,20 @@ def save_as_chunk(self, images: Iterator[tuple[Image.Image|io.IOBase|str, str, s
                 ext = os.path.splitext(path)[1].replace('.', '')
 
                 if self._dimension == DimensionType.DIM_2D:
-                    # current version of Pillow applies exif rotation immediately when TIFF image opened
-                    # and it removes rotation tag after that
-                    # so, has_exif_rotation(image) will return False for TIFF images even if they were actually rotated
-                    # and original files will be added to the archive (without applied rotation)
-                    # that is why we need the second part of the condition
-                    if isinstance(image, Image.Image) and (
-                        has_exif_rotation(image) or image.format == 'TIFF'
-                    ):
+                    if isinstance(image, Image.Image):
```

Comment:
This seems to change the chunk writing logic quite a bit. Previously, this would only re-encode images if they had to be re-oriented; now it always does it. In effect, the `else` branch of this `if` will never be reached now.

Is this really what we want? Seems like this will waste CPU time on unnecessary re-encoding, and cause quality degradation if the target format is lossy.

Reply:
Oh yeah, I wanted to optimize by rotating only once when loading the image and indeed the logic of the function changed, I reverted this change.